packages/sapi/src/helpers/getDryRunCall.ts

```diff
@@ -1,35 +1,49 @@
-import { polkadot, polkadotAssetHub, PolkadotRuntimeOriginCaller } from "@polkadot-api/descriptors"
-import { log } from "extension-shared"
-import { TypedApi } from "polkadot-api"
+import type {
+  polkadot,
+  polkadotAssetHub,
+  PolkadotRuntimeOriginCaller,
+} from "@polkadot-api/descriptors"
+import { Enum } from "polkadot-api"
 
+import log from "../log"
 import { DecodedCall } from "../types"
 import { getDispatchErrorMessage } from "./errors"
 import { getRuntimeCallResult } from "./getRuntimeCallResult"
 import { isApiAvailable } from "./isApiAvailable"
 import { Chain } from "./types"
 
-type DryRunRefChain = typeof polkadot | typeof polkadotAssetHub
-type DryRunResult = Awaited<
-  ReturnType<TypedApi<DryRunRefChain>["apis"]["DryRunApi"]["dry_run_call"]>
->
+type DryRunResult = (
+  | typeof polkadot
+  | typeof polkadotAssetHub
+)["descriptors"]["apis"]["DryRunApi"]["dry_run_call"][1]
 
-export const getDryRunCall = async (
+export const getDryRunCall = async <T>(
   chain: Chain,
   from: string,
   decodedCall: DecodedCall<unknown>,
-) => {
-  const stop = log.timer("[sapi] getDryRun")
+): Promise<
+  | {
+      available: boolean
+      data: null
+      ok?: undefined
+      errorMessage?: undefined
+    }
+  | {
+      available: boolean
+      data: T
+      ok: boolean
+      errorMessage: string | null
+    }
+> => {
+  log.log(`[sapi] getDryRun begin: ${Date.now()}`)
   try {
     if (!isApiAvailable(chain, "DryRunApi", "dry_run_call"))
       return {
         available: false,
         data: null,
       }
 
-    const origin = PolkadotRuntimeOriginCaller.system({
-      type: "Signed",
-      value: from,
-    })
+    const origin: PolkadotRuntimeOriginCaller = Enum("system", Enum("Signed", from))
```

Comment:
We should always use `import type` for `@polkadot/api-descriptors` where possible, so that we don't end up importing the full metadata for each chain into the extension runtime environment.